tensorflow/core/platform/cloud/retrying_file_system.cc

```diff
@@ -22,8 +22,10 @@ namespace tensorflow {
 
 namespace {
 
-// In case of failure, every call will be retried kMaxAttempts-1 times.
-constexpr int kMaxAttempts = 4;
+// In case of failure, every call will be retried kMaxAttempts times.
+constexpr int kMaxAttempts = 3;
```

Comment:
Ok, I think I got it. It looks like kMaxAttempts has become a "retry count" instead of an "attempt count" that it used to be. So this needs to be fixed to be made consistent again. 

One way would be to rename it to kMaxRetries and rename 'attempt' to 'retry' respectively, keeping the constant value '3', the comment and the rest of the code as it is now.

Another, maybe a simpler way, would be to replace "attempt >= kMaxAttempts" with "attempt >= kMaxAttempts -1" to solve the "off by one" error.  And then the lines 25-26 can be reverted back to the previous state.

What do you think?